gms/impl/src/main/java/com/linkedin/metadata/resources/dataset/Datasets.java

```diff
@@ -145,14 +158,30 @@ protected Dataset toValue(@Nonnull DatasetSnapshot snapshot) {
   @Nonnull
   protected DatasetSnapshot toSnapshot(@Nonnull Dataset dataset, @Nonnull DatasetUrn datasetUrn) {
     final List<DatasetAspect> aspects = new ArrayList<>();
-    if (dataset.hasProperties()) {
+    if (dataset.getProperties() != null) {
       aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, getDatasetPropertiesAspect(dataset)));
     }
-    if (dataset.hasDeprecation()) {
+    if (dataset.getDeprecation() != null) {
       aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, dataset.getDeprecation()));
     }
-
-    aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, new Status().setRemoved(dataset.isRemoved())));
+    if (dataset.getInstitutionalMemory() != null) {
+      aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, dataset.getInstitutionalMemory()));
+    }
+    if (dataset.getOwnership() != null) {
+      aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, dataset.getOwnership()));
+    }
+    if (dataset.getSchemaMetadata() != null) {
+      aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, dataset.getSchemaMetadata()));
+    }
+    if (dataset.getStatus() != null) {
+      aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, dataset.getStatus()));
+    }
+    if (dataset.getUpstreamLineage() != null) {
+      aspects.add(ModelUtils.newAspectUnion(DatasetAspect.class, dataset.getUpstreamLineage()));
+    }
+    if (dataset.hasRemoved()) {
+      aspects.add(DatasetAspect.create(new Status().setRemoved(dataset.isRemoved())));
+    }
```

Comment:
This is interesting. Not sure if we have the same issue in internal code base. Seems like we both use Status aspect itself to get removed and also we have a separate removed field the model. Is this intentional? What if they're conflicting with each other?

Reply:
Until the FE move away from this deprecated field, we'll need to populate both the Status aspect as well as this field, and use the former as SOT.

Reply:
Thanks for the input @mars-lan. I would like to understand why populating `Status` aspect is preferred over `removed` field in the `VALUE` model of datasets. Is it because the `removed` field has a default value which might not be desired? had `removed` been an optional field it would have worked?

Reply:
`removed` is a legacy field that we should drop eventually. I'm not sure if FE is still dependent on this field or pulling from `Status` already.

Reply:
Marked the `removed` field as deprecated. Also looks like FE is dependent on `removed` field, changed to using `Status` field instead.